DEVELOPMENT.md

```diff
@@ -1,87 +1,78 @@
 ## Install BentoML from source
 
-Ensure you have git, python and pip installed, BentoML supports Python 3.7+
+Make sure to have [Git](https://git-scm.com/) and [Python3.7+](https://www.python.org/downloads/) installed.
+
+Optionally, make sure to have [GNU Make](https://www.gnu.org/software/make/) available on your system if you aren't using UNIX-based system for better developer experience.
+If you don't want to use `make` then refers to [Makefile](./Makefile) for specific commands on given make target.
+
+We are also using [docker](https://www.docker.com/) for our code style scripts that will be mentioned below for better devex.
 
 ```bash
 python --version
-```
 
-```bash
 pip --version
 ```
 
-
 Clone the source code from BentoML's GitHub repository:
 ```bash
-git clone https://github.com/bentoml/BentoML.git
-cd BentoML
+git clone https://github.com/bentoml/BentoML.git && cd BentoML
 ```
 
 Install BentoML with pip in `editable` mode:
 ```bash
-pip install --editable .
+make install-local
```

Comment:
We can use `poetry install` now I believe, if we want to move in that direction.

Reply:
I will add a docs that if users want to use poetry then they can do so as well